lib/src/local_working_copy.rs

```diff
@@ -671,6 +684,56 @@ impl TreeState {
         Ok(())
     }
 
+    fn save_only_clock(&mut self) -> Result<(), TreeStateError> {
+        let mut temp_file = NamedTempFile::new_in(&self.state_path).unwrap();
+        let target_path = self.state_path.join("tree_state");
+
+        // If the same non-repeated field appears multiple times, only the last will be used:
+        // https://protobuf.dev/programming-guides/encoding/#last-one-wins
+        // Reuse the existing serialisation of the proto in the current state to avoid
+        // serialising non-changed fields.
+        let bytes_written = std::fs::copy(&target_path, temp_file.path()).map_err(|err| {
+            TreeStateError::CopyTreeState {
+                read_path: target_path.clone(),
+                write_path: temp_file.path().to_owned(),
+                source: err,
+            }
+        })?;
+
+        let file = temp_file.as_file_mut();
+        file.seek(std::io::SeekFrom::Start(bytes_written))
+            .map_err(|err| TreeStateError::WriteTreeState {
+                path: self.state_path.clone(),
+                source: err,
+            })?;
+
+        let proto = crate::protos::working_copy::TreeState {
+            watchman_clock: self.watchman_clock.clone(),
+            ..Default::default()
+        };
```

Comment:
nit: I assume this never serializes fields other than `watchman_clock` because `Default` values can be omitted in protobuf, and the serialization should be skipped by prost.

Can you add a short comment about that?